buildrunner/config/jinja_context.py

```diff
@@ -0,0 +1,81 @@
+"""
+Copyright 2024 Adobe
+All Rights Reserved.
+
+NOTICE: Adobe permits you to use, modify, and distribute this file in accordance
+with the terms of the Adobe license agreement accompanying it.
+"""
+
+import codecs
+import copy
+import datetime
+import re
+from io import StringIO
+from typing import Callable
+
+import jinja2
+
+from buildrunner.utils import load_config
+
+
+def read_yaml_file(
+    env: dict,
+    log_generated_file_method: Callable[[bool, str, str], None],
+    log_file: bool,
+    filename: str,
+):
+    """
+    Reads a file in the local workspace as Jinja-templated YAML and returns the contents.
+    Throws an error on failure.
+    """
+    with codecs.open(filename, "r", encoding="utf-8") as _file:
+        jtemplate = jinja2.Template(_file.read())
+    context = copy.deepcopy(env)
+    file_contents = jtemplate.render(context)
+    log_generated_file_method(log_file, filename, file_contents)
+    return load_config(StringIO(file_contents), filename)
+
+
+def strftime(_build_time: int, _format="%Y-%m-%d", _ts=None):
+    """
+    Format the provided timestamp. If no timestamp is provided, build_time is used
+    :param _build_time: The build timestamp. This is bound with functools.partial and is not required to pass in.
+    :param _format: Format string - default "%Y-%m-%d"
+    :param _ts: Timestamp to format - default self.build_time
+    :return: Formatted date/time string
+    """
+    if _ts is None:
+        _ts = _build_time
+    _date = datetime.datetime.fromtimestamp(_ts)
+    return _date.strftime(_format)
+
+
```

Comment:
Alot of these methods just feel like filler, is there a benifit to these instead of just calling the methods directly? maybe less imports directly ?

And why are we raising an Exception for jinja but then it's just a regular exception?

Reply:
These are all just utility methods/filters for the jinja context. I don't know how we would call them directly, since they are just passed into the jinja context as available methods/filters.

Reply:
The exception is raised within a jinja template, it's not for jinja itself. i.e. nothing special about it, just a utility method to be used in jinja templates.